nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkManagerImpl.java

```diff
@@ -419,6 +381,20 @@ public void afterStop(ComponentManager mgr, boolean isStandby) {
         }
     }
 
+    protected void index() {
+        List<WorkQueueDescriptor> descriptors = getDescriptors(QUEUES_EP);
+        descriptors.forEach(d -> d.categories.forEach(c -> {
+            categoryToQueueId.putIfAbsent(c, d.getId());
+            categoryToQueueId.computeIfPresent(c, (k, v) -> {
```

Comment:
You should inverse the two lines above to avoid executing the `BiFunction` on each first time.

Reply:
nice one !